### DISCUSSION TITLE
Requested Forecast Grid has Expired

### DISCUSSION THREAD

**Discussion by kconrad37** [weather-gov/api#371] [Q&A]
The company I am working for was using a now outdated version of this service until it finally stopped working around the top of the year (give or take). Since then I have rewritten our program to use this api service, and for the most part it seems to work. 

However our results seem to be riddled with 503 errors, and looking through our Error Log today, they all seem to be associated with cwa "BOX."

When I run calls using this cws (office?) using Postman, I keep getting the following response:

{
    "correlationId": "3ed8bace-d53b-4d32-8175-68d396faffd3",
    "title": "Forecast Grid Expired",
    "type": "https://api.weather.gov/problems/ForecastGridExpired",
    "status": 503,
    "detail": "The requested forecast grid was issued 2020-02-11T02:50:40+00:00 and has expired.",
    "instance": "https://api.weather.gov/requests/3ed8bace-d53b-4d32-8175-68d396faffd3"
}

(correlationIds and instances likely to differ, if I'm not mistaken)

A few examples:

https://api.weather.gov/gridpoints/BOX/10,39/forecast/hourly
https://api.weather.gov/gridpoints/BOX/12,34/forecast
https://api.weather.gov/gridpoints/BOX/10,39/forecast/hourly

I just ran this gridpoint call (https://api.weather.gov/points/41.6578977,-72.6797965) to try a new link directly from it (https://api.weather.gov/gridpoints/BOX/21,35/forecast), which also gives an error.

{
    "correlationId": "36eb9a42-990d-4ca8-a24a-cd0c67985903",
    "title": "Forecast Grid Expired",
    "type": "https://api.weather.gov/problems/ForecastGridExpired",
    "status": 503,
    "detail": "The requested forecast grid was issued 2020-02-11T02:50:40+00:00 and has expired.",
    "instance": "https://api.weather.gov/requests/36eb9a42-990d-4ca8-a24a-cd0c67985903"
}

Unfortunately I don't know what it means when a forecast grid has expired so I don't know how I'm supposed to fix this and provide accurate weather information.

This is a java program and it uses javax.net.ssl.HttpsURLConnection and java.net.URL:

URL url = new URL("https://api.weather.gov/gridpoints/BOX/21,35/forecast"); //sub in any link
HttpsURLConnection httpsConn = (HttpsURLConnection)url.openConnection();
httpsConn.setRequestMethod(HttpMethod.GET);
httpsConn.setRequestProperty("Content-Type", "application/json");
httpsConn.setConnectTimeout((1000 * 60));
httpsConn.setReadTimeout((1000 * 60));

StringBuffer response = new StringBuffer();			
BufferedReader in = new BufferedReader(new InputStreamReader(httpsConn.getInputStream()));
String inputLine;
while((inputLine = in.readLine()) != null){
	response.append(inputLine);
}
in.close();

I would really appreciate some clarification on what it means when a forecast grid expires and if there's any way at all to stop this from happening.

**Comment by scadergit** [comment]
This is an operational outage, not a bug in your software or the API. The forecast girds were literally not getting from the office to the API, and the current grids started to expire out. I believe this has been resolved. Please open an issue with TOC (instructions in the documentation on this site) if the issue persists.

**Comment by kconrad37** [comment]
> This is an operational outage, not a bug in your software or the API. The forecast girds were literally not getting from the office to the API, and the current grids started to expire out. I believe this has been resolved. Please open an issue with TOC (instructions in the documentation on this site) if the issue persists.

Thank you for your response. I was not sure if this was a bug or an outage issue, so it felt safer to call it a bug. My apologies if I posted in the wrong location.

**Comment by scadergit** [comment]
No worries, always good to hear from the community using this service. We'll take this as feedback to make error messages more easily understood.

**Comment by arturtle** [comment]
I know this is closed, but thanks for the update on StackOverflow. I'm new to javascript and I thought I had accidentally broken something until I came across your posts.
